090-integrate-bridge-with-metrics-reporter.md

```diff
@@ -0,0 +1,113 @@
+In order to provide a consistent user experience, the HTTP Bridge should align with all the other components in terms of metrics configuration. 
+
+## Proposal
+
+As specified in SIP-064, a new metrics config type called `strimziMetricsReporter` will be added to `metricsConfig` schema along with its configuration.
+Only the Strimzi Metrics Reporter's `prometheus.metrics.reporter.allowlist` property will be configurable by users when using the Cluster Operator.
+This configuration is a comma separated list of regex patterns that is used by the plugin to filter metrics to collect.
+All other plugin configuration properties will be locked down, including the port that will be set to 9404 to avoid conflicts with other ports.
+
+```sh
+spec:
+  kafka:
+    metricsConfig:
+      type: strimziMetricsReporter
+      values:
+        allowList:
+          - "kafka_log.*"
+          - "kafka_network.*"
+```
+
+### HTTP Bridge
+
+The HTTP Bridge project will be updated to support multiple types of metrics configurations.
+A new `bridge.metrics` property will be available, and will only accept `jmxPrometheusExporter` and `strimziMetricsReporter`.
+Any other value will raise an error and the application will fail to start with an appropriate error message.
+
+When running in standalone mode with `strimziMetricsReporter`, the user will be able to configure any Strimzi Metrics Reporter property using the `kafka.` prefix.
+At a minimum `kafka.metric.reporters=io.strimzi.kafka.metrics.KafkaPrometheusMetricsReporter` and `kafka.prometheus.metrics.reporter.listener.enable=false` will be needed.
+The following example will be provided as a comment in the default properties file.
+
+```sh
+bridge.metrics=strimziMetricsReporter
+kafka.metric.reporters=io.strimzi.kafka.metrics.KafkaPrometheusMetricsReporter
+kafka.prometheus.metrics.reporter.listener.enable=false
+kafka.prometheus.metrics.reporter.allowlist=.*
+```
+
+The `KAFKA_BRIDGE_METRICS_ENABLED` env var will be deprecated and removed in a future release.
+When set, the user will get a warning suggesting to use the new `bridge.metrics` configuration.
+In case they are both set, the `bridge.metrics` configuration will take precedence over `KAFKA_BRIDGE_METRICS_ENABLED`.
+
+The `MetricsReporter` class will be updated to also include a new `StrimziCollectorRegistry` that will work similarly to the `JmxCollectorRegistry`.
+The `StrimziCollectorRegistry` will include a reference to `PrometheusRegistry.defaultRegistry`, which is the same instance used by the Strimzi Metrics Reporter to collect metrics.
+
+The `kafka_bridge_config_generator.sh` script is used to generate the image configuration based on environment variables.
+This script will be updated to also include `bridge.metrics` and related configurations (see the following section).
+
+The Bridge will also try to load the JMX Exporter configuration file from the path specified by the `bridge.metrics.jmx.exporter.config.path` property.
```

Comment:
This sounds to be an addition to the enablement of the Strimzi metrics reporter.
I mean, we could add the Strimzi metrics reporter and leaving the current JMX reporter still working with the hard-coded configuration. I am not against it but if we want to have both things we should make it clearer that this is a new addition and not needed for the Strimzi metrics reporter which is the main goal of the proposal.

Reply:
Initially, this wasn't part of the proposal, but then I realized that the JMX Exporter configuration is required when using the shared `metricsConfig` schema, and we have it in all the metrics examples. Additionally, I think it adds a useful feature for the standalone Bridge users. Wdyt?

Reply:
I am fine with it, but let's make it clear it's an addition not something needed for the Strimzi metrics reporter support.

Reply:
Anyway when you said "it's required" ... don't we have any way to fall back using the default "hard coded" configuration? I think it's mentioned somewhere else in the proposal, or?

Reply:
> let's make it clear it's an addition not something needed for the Strimzi metrics reporter support

Ok, will do.

> don't we have any way to fall back using the default "hard coded" configuration? 

Correct. Let me fix.